cli/src/pcluster/resources/compute_node/user_data.sh

```diff
@@ -151,6 +151,18 @@ write_files:
         done;
         export HOME="${!HOME_BAK}"
       }
+
+      function publish_startup_time
+      {
+        StartTime=$1
+        EndTime=$2
+        METRIC_NAMESPACE="ParallelCluster"
+        METRIC_NAME="StartupTime"
+        INSTANCE_ID=$(cat /etc/parallelcluster/slurm_plugin/slurm_node_spec.json | jq -r .instance_id)
+        INSTANCE_TYPE=$(cat /etc/parallelcluster/slurm_plugin/slurm_node_spec.json | jq -r .'compute."instance-type"')
+        aws cloudwatch put-metric-data --namespace $METRIC_NAMESPACE --metric-name $METRIC_NAME --value $(($EndTime-$StartTime)) --dimensions "InstanceID=${!INSTANCE_ID},InstanceType=${!INSTANCE_TYPE},ClusterName=${ClusterName}" --region "${AWS::Region}"
```

Comment:
Did you test what happen if the node does not have the right IAM permissions to do the "put-metric-data" ?

Reply:
When the node does not have the IAM permission to do "put-metric-data" this command will simply not be executed showing warning message "Access Denied". Besides this everything else is working.